sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVOptions.scala

```diff
@@ -90,6 +90,7 @@ class CSVOptions(
   val quote = getChar("quote", '\"')
   val escape = getChar("escape", '\\')
   val comment = getChar("comment", '\u0000')
+  val lineSeparator = parameters.getOrElse("lineSeparator", "\n")
```

Comment:
Ah, I think I understand now, thanks. 

Maybe I'm overlooking it, but I don't see the `wholeFile` option being used in `CsvOptions` at all at the moment (nor `multiline`) ?

I'll take a further look at fixing this up considering the dependency on `LineRecordReader` (which only supports the two line separators you mentioned)